### PR TITLE
clean domain server code

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

File: domain-server/src/DomainServer.cpp [hunk 1]
```diff
@@ -97,40 +96,42 @@ DomainServer::DomainServer(int argc, char* argv[]) :
     // make sure we hear about newly connected nodes from our gatekeeper
     connect(&_gatekeeper, &DomainGatekeeper::connectedNode, this, &DomainServer::handleConnectedNode);
 
-    if (optionallyReadX509KeyAndCertificate() && optionallySetupOAuth()) {
-        // we either read a certificate and private key or were not passed one
-        // and completed login or did not need to
+    // we either read a certificate and private key or were not passed one
+    // and completed login or did not need to;
+    // if we were passed partial information, we fail and need to quit
+    if (!(optionallyReadX509KeyAndCertificate() && optionallySetupOAuth())) {
```
Comment (zzmp) [inline domain-server/src/DomainServer.cpp]: The logic that was nested in here was not dependent on these things, so I took it out.

File: domain-server/src/DomainServer.cpp [hunk 2]
```diff
@@ -172,8 +174,6 @@ bool DomainServer::optionallyReadX509KeyAndCertificate() {
 
     } else if (!certPath.isEmpty() || !keyPath.isEmpty()) {
         static const QString MISSING_CERT_ERROR_MSG = "Missing certificate or private key. domain-server will now quit.";
-        static const int MISSING_CERT_ERROR_CODE = 3;
```
Comment (zzmp) [inline domain-server/src/DomainServer.cpp]: I moved all of the error codes to the header for more readable self-documentation.

**Comment by zzmp** [comment]
This PR is entirely cosmetic.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 04547f47416ac6542c0ffa1623f7128bb9db8be1 available at: https://gist.github.com/948ab0df2d8b4ea7fd15e73ad32b6975<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7981.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7981.dmg

**Comment by ZappoMan** [comment]
This appears to touch the logic flow quite a bit... we need a solid test plan to make sure nothing got accidentally broken. Each feature that got touched needs to be tested for correctness.

**Comment by zzmp** [comment]
#### Testing

Test plan is WIP.

This test plan covers the domain-server, *not* Interface.

Behaviors that were changed:
- Quitting (changed codepath)
  - Ensure that logging during quit and restart has not changed. You can restart from the settings page.
- X509 key/cert and oauth (changed codepath)
  - Ensure that startup doesn't hang.
- `--get-temp-name` switch
  - Start up an unnamed domain-server with the `--get-temp-name` switch and you should see "Received new temporary domain name" in the logs.
- Heartbeats to Metaverse
  - Both domains with full and ip automatic network settings should regularly send heartbeats to the metaverse
- Assignment configuration
  - Start a domain-server with the `--config-` option and you should see...

**Comment by zzmp** [comment]
Not QA ready yet, I still need to figure out how the last option in the test plan is used.

**Comment by ZappoMan** [comment]
oh sorry @zzmp -

**Pull request closed without merging**